### PR TITLE
fix: additional changes to address mypy 'call-overload'; adding cron for ci

### PR DESCRIPTION
**Pull request by aorumbayev**
## Proposed Changes

  - Additional tweaks to address mypy >=1.11.0 warnings
  - Adding a cron trigger to re run CI each monday

### CODE REVIEW

**Review by neilcampbell — APPROVED** [review]

File: examples/helloworld/client.py [hunk 1]
```diff
@@ -128,7 +128,7 @@ def _as_dict(data: typing.Any, *, convert_all: bool = True) -> dict[str, typing.
     if not dataclasses.is_dataclass(data):
         raise TypeError(f"{data} must be a dataclass")
     if convert_all:
-        result = typing.cast(dict[str, typing.Any], dataclasses.asdict(data))
+        result = dataclasses.asdict(data) # type: ignore[call-overload]
```
Comment (cusma) [inline examples/helloworld/client.py:131]: nit: add an extraspace before `# type: ignore[call-overload]` to comply with PEP 8 E261 (at least two spaces before inline comment)

File: examples/lifecycle/client.py [hunk 2]
```diff
@@ -174,7 +174,7 @@ def _as_dict(data: typing.Any, *, convert_all: bool = True) -> dict[str, typing.
     if not dataclasses.is_dataclass(data):
         raise TypeError(f"{data} must be a dataclass")
     if convert_all:
-        result = typing.cast(dict[str, typing.Any], dataclasses.asdict(data))
+        result = dataclasses.asdict(data) # type: ignore[call-overload]
```
Comment (cusma) [inline examples/lifecycle/client.py:177]: nit: same, comply with PEP 8 E261

File: examples/minimal/client.py [hunk 3]
```diff
@@ -90,7 +90,7 @@ def _as_dict(data: typing.Any, *, convert_all: bool = True) -> dict[str, typing.
     if not dataclasses.is_dataclass(data):
         raise TypeError(f"{data} must be a dataclass")
     if convert_all:
-        result = typing.cast(dict[str, typing.Any], dataclasses.asdict(data))
+        result = dataclasses.asdict(data) # type: ignore[call-overload]
```
Comment (cusma) [inline examples/minimal/client.py:93]: nit: same, comply with PEP 8 E261

File: examples/state/client.py [hunk 4]
```diff
@@ -464,7 +464,7 @@ def _as_dict(data: typing.Any, *, convert_all: bool = True) -> dict[str, typing.
     if not dataclasses.is_dataclass(data):
         raise TypeError(f"{data} must be a dataclass")
     if convert_all:
-        result = typing.cast(dict[str, typing.Any], dataclasses.asdict(data))
+        result = dataclasses.asdict(data) # type: ignore[call-overload]
```
Comment (cusma) [inline examples/state/client.py:467]: nit: same, comply with PEP 8 E261

File: examples/voting/client.py [hunk 5]
```diff
@@ -300,7 +300,7 @@ def _as_dict(data: typing.Any, *, convert_all: bool = True) -> dict[str, typing.
     if not dataclasses.is_dataclass(data):
         raise TypeError(f"{data} must be a dataclass")
     if convert_all:
-        result = typing.cast(dict[str, typing.Any], dataclasses.asdict(data))
+        result = dataclasses.asdict(data) # type: ignore[call-overload]
```
Comment (cusma) [inline examples/voting/client.py:303]: nit: same, comply with PEP 8 E261

File: src/algokit_client_generator/generator.py [hunk 6]
```diff
@@ -190,7 +190,7 @@ def _as_dict(data: typing.Any, *, convert_all: bool = True) -> dict[str, typing.
     if not dataclasses.is_dataclass(data):
         raise TypeError(f"{data} must be a dataclass")
     if convert_all:
-        result = typing.cast(dict[str, typing.Any], dataclasses.asdict(data))
+        result = dataclasses.asdict(data) # type: ignore[call-overload]
```
Comment (cusma) [inline src/algokit_client_generator/generator.py:193]: nit: same, comply with PEP 8 E261

**Pull request merged**